compiler/src/iree/compiler/Dialect/Flow/IR/FlowOps.td

```diff
@@ -1768,4 +1768,43 @@ def FLOW_CollectiveReduceScatterOp : FLOW_Op<"collective.reduce_scatter", [
   ];
 }
 
+def FLOW_CollectivePermuteOp : FLOW_Op<"collective.permute", [
```

Comment:
This op is not a collective permute. It is `sendrecv`, which does send and recv in a single operation, which is neat because normally we send data asynchronously and use a blocking recv. OTHO, collective permute is a concurrent execution of multiple send and recv.

Another thing to consider is that lowering collective permute into multiple sendrecv at the flow level may hurt the performance since they are not in a single NCCL group start/end. We may want to split a collective permute into multiple sendrecvs when we have more advanced compiler analysis and optimization to increase the computation and communication overlapping, but the effort is not scoped out yet. Until then, it would be better to put the sendrecv ops under the hal level to put them in a single nccl group.

Reply:
According to the stablehlo spec, each rank can only send to one rank (or none), and receive from one rank (or none). There will never be more than one send/recv per rank, so we are already grouping all of communications.

Reply:
Oh, yeah, I realize that and removed my comment about the nccl group. Sorry for the confusing comment.

Reply:
No worries!